shap/explainers/_deep/deep_tf.py

```diff
@@ -130,33 +130,35 @@ def __init__(self, model, data, session=None, learning_phase_flags=None):
         # if no learning phase flags were given we go looking for them
         # ...this will catch the one that keras uses
         # we need to find them since we want to make sure learning phase flags are set to False
-        if learning_phase_flags is None:
-            self.learning_phase_ops = []
-            for op in self.graph.get_operations():
-                if 'learning_phase' in op.name and op.type == "Const" and len(op.outputs[0].shape) == 0:
-                    if op.outputs[0].dtype == tf.bool:
-                        self.learning_phase_ops.append(op)
-            self.learning_phase_flags = [op.outputs[0] for op in self.learning_phase_ops]
+        if self.graph is not None:
+            if learning_phase_flags is None:
+                self.learning_phase_ops = []
+                for op in self.graph.get_operations():
+                    if 'learning_phase' in op.name and op.type == "Const" and len(op.outputs[0].shape) == 0:
+                        if op.outputs[0].dtype == tf.bool:
+                            self.learning_phase_ops.append(op)
+                self.learning_phase_flags = [op.outputs[0] for op in self.learning_phase_ops]
+            else:
+                self.learning_phase_ops = [t.op for t in learning_phase_flags]
         else:
-            self.learning_phase_ops = [t.op for t in learning_phase_flags]
+            self.learning_phase_ops = self.learning_phase_flags = []
 
         # save the expected output of the model
         # if self.data is a function, set self.expected_value to None
-        if (hasattr(self.data, '__call__')):
+        if hasattr(self.data, '__call__'):
             self.expected_value = None
         else:
             if self.data[0].shape[0] > 5000:
                 warnings.warn("You have provided over 5k background samples! For better performance consider using smaller random sample.")
             if not tf.executing_eagerly():
                 self.expected_value = self.run(self.model_output, self.model_inputs, self.data).mean(0)
             else:
-                if type(self.model)is tuple:
-                    sel.fModel(cnn.inputs, cnn.get_layer(theNameYouWant).outputs)
```

Comment:
I do think this line is completely unnecessary, even if it was to work?

Reply:
Odd, yes it seems like a left over typo. Thanks!